File: 算法训练营/20-字符串算法/058/迭代法.py

#
# @lc app=leetcode.cn id=58 lang=python3
#
# [58] 最后一个单词的长度
#

# @lc code=start


class Solution:
    def lengthOfLastWord(self, s: str) -> int:
        pre_len = 0
        cur_len = 0  # 当前字符串长度

        for i in s:
            if i != " ":
                cur_len += 1
                continue
            pre_len = cur_len if cur_len > 0 else pre_len
            cur_len = 0

        return cur_len if cur_len > 0 else pre_len
# @lc code=end

File: LeetCode/0027/迭代法.go
/*
 * @lc app=leetcode.cn id=27 lang=rust
 *
 * [27] 移除元素
 */

// @lc code=start
impl Solution {
    pub fn remove_element(nums: &mut Vec<i32>, val: i32) -> i32 {
        let l: usize = nums.len(); 
        let mut count: usize = 0;
        for i in 0..l {
            if nums[i] == val {
                count = count + 1;
            } else if i != i-count {
                nums[i-count] = nums[i]
            }
        }
        (l - count) as i32
    }
}
// @lc code=end
